README.md

```diff
@@ -1,27 +1,102 @@
 # The Graylog documentation
 [![Build Status](https://travis-ci.org/Graylog2/documentation.svg?branch=2.2)](https://travis-ci.org/Graylog2/documentation)
 
+## Architecture
+
+This is the repository that is used to create [docs.graylog.org](http://docs.graylog.org) running at [readthedocs](https://readthedocs.org/). After changes are pushed to the specific branches, the new pages are built automatically. When the build runs without error, the new pages are available after some minutes.
+
+The typical workflow to make changes to the documentation and preview them before you push the changes to this repository needs the following preparation. 
+
+- clone the repository (to your workplace)
+- initialize the virtual environment for python
+- install the required python packages
+
+After the above is done, changes can be made and previewed with the following
+
+- create git branch for your changes
+- change in the virtual environment
+- make the modifications/additions
+- run the build and check for errors
+- push the branch to GitHub and create a pull request
+
+Now a review of the changes is needed and finally, it will be merged, by the reviewer into the branch/version of the documentation you created the pull request for. Should the change be available in other versions of the documentation this should be written in the pull request that the reviewer can push this to the specific versions too.
```

Comment:
Now a review of the changes is needed and, finally, it will be merged by the reviewer